# Probability Intuition Pt. 2

In [1]:
import pandas as pd

In [2]:
# synthetic dataset simulating dice combinations and frequencies
dice_df = pd.read_csv('dice.csv')

In [3]:
dice_df.head()

dice_1  dice_2
0       1       1
1       1       2
2       1       3
3       1       4
4       1       5

## Probabilities of Rolling Two Dice

In [4]:
# probability of rolling a 1 with dice 1 and another 1 with dice 2
desired_outcomes = len(dice_df[(dice_df['dice_1']==1) & (dice_df['dice_2']==1)])
possible_outcomes = len(dice_df) # 36 - total number of combinations from rolling 2 dice

# probability
desired_outcome / possible_outcomes

0.027777777777777776

There is a 2.7% of rolling a '1' with dice 1 and another '1' with dice 2 *(if both dice are fair)*.

In [5]:
# probability of rolling an even number for each dice
desired_outcomes = len(dice_df[(dice_df['dice_1']%2==0) & (dice_df['dice_2']%2==0)])
possible_outcomes = len(dice_df) # 36

# probability
desired_outcome / possible_outcomes

0.25

We have a 25% chance of rolling an even number with both dice *(if both dice are fair)*.

In [6]:
# desired outcome - rolling a 1 with dice 1 and an odd number with dice 2
desired_outcomes = len(dice_df[(dice_df['dice_1']==1) & (dice_df['dice_2']%2!=0)])
# possible outcome - all the possible dice combinations
possible_outcomes = len(dice_df)

# probability
desired_outcome / possible_outcomes

0.08333333333333333

There is an 8.333% chance of rolling a '1' with dice 1 and an odd number with dice 2 *(if both dice are fair)*.

In [7]:
# desired outcome - rolling a 2 with dice 1 regardless of the outcome of dice 2
desired_outcome = len(dice_df[dice_df['dice_1']==2])
# possible outcomes - all the possible dice combinations
possible_outcomes = len(dice_df)

# probability
desired_outcome / possible_outcomes

0.16666666666666666

There is a 16.66% chance if rolling a '2' with dice 1 regardless of the outcomes from dice 2 *(if both dice are fair)*.